Add KiKit and dependencies

hesiod opened the pull request:
###### Description of changes

This PR adds [KiKit](https://github.com/yaqwsx/KiKit) as well as its dependencies (all Python packages):
* `pcbnew-transition`
* `pybars3`
* `pymeta3`
* `solidpython`
* `euclid3`

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

hesiod committed:
python3Packages.euclid3: init at 0.01

hesiod committed:
python3Packages.solidpython: init at 1.1.3

hesiod committed:
python3Packages.pymeta3: init at 0.5.1

hesiod committed:
python3Packages.pybars3: init at 0.9.7

hesiod committed:
python3Packages.pcbnew-transition: init at 0.3.4

hesiod committed:
python3Packages.kikit: init at 1.3.0

hesiod commented:
Result of `nixpkgs-review pr 211829` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>10 packages built:</summary>
  <ul>
    <li>python310Packages.euclid3</li>
    <li>python310Packages.pcbnew-transition</li>
    <li>python310Packages.pybars3</li>
    <li>python310Packages.pymeta3</li>
    <li>python310Packages.solidpython</li>
    <li>python311Packages.euclid3</li>
    <li>python311Packages.pcbnew-transition</li>
    <li>python311Packages.pybars3</li>
    <li>python311Packages.pymeta3</li>
    <li>python311Packages.solidpython</li>
  </ul>
</details>

hesiod commented:
For some reason the addition of `kikit` in `all-packages.nix` didn't get picked up previously (I probably forgot to `git add` it), that's why the rebuild number went up

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/1923

SuperSandro2000 commented on pkgs/development/python-modules/solidpython/default.nix:
```diff
@@ -0,0 +1,42 @@
+  '';
+
+  pythonImportsCheck = [ "solid" ];
+
```
are we running those tests https://github.com/SolidCode/SolidPython/tree/master/solid/test ?

hesiod replied:
Good catch, I added `unittestCheckHook` (and a minor patch) in order to run those tests

SuperSandro2000 commented on pkgs/development/python-modules/pymeta3/default.nix:
```diff
@@ -0,0 +1,31 @@
+
+  checkInputs = [ twisted ];
+
+  # Tests broken on Python 3 (upstream bugs)
+  doCheck = false;
```
FYI python2 is unuspported

SuperSandro2000 replied:
We should think about removing the package if it does not support python3

SuperSandro2000 commented on pkgs/development/python-modules/pybars3/default.nix:
```diff
@@ -0,0 +1,34 @@
+    hash = "sha256-ashH6QXlO5xbk2rxEskQR14nv3Z/efRSjBb5rx7A4lI=";
+  };
+
+  preCheck = ''rm tests.py'';
+  checkInputs = [ unittestCheckHook ];
+
+  propagatedBuildInputs = [ pymeta3 ];
```
```suggestion
  propagatedBuildInputs = [ pymeta3 ];

  checkInputs = [ unittestCheckHook ];

  preCheck = ''
    rm tests.py
  '';
```

SuperSandro2000 commented on pkgs/development/python-modules/kikit/default.nix:
```diff
@@ -0,0 +1,66 @@
+    wcwidth
+  ];
+
+  pythonImportsCheck = [ pname ];
```
```suggestion
  pythonImportsCheck = [ "kikit" ];
```

SuperSandro2000 commented on pkgs/development/python-modules/kikit/default.nix:
```diff
@@ -0,0 +1,66 @@
+  version = "1.2.0";
+  format = "setuptools";
+
+  # Sources via fetchPypi don't contain tests
```
```suggestion
```
pretty normal and not worth a comment

SuperSandro2000 commented on pkgs/development/python-modules/euclid3/default.nix:
```diff
@@ -0,0 +1,24 @@
+    hash = "sha256-JbgnpXrb/Zo/qGJeQ6vD6Qf2HeYiND5+U4SC75tG/Qs=";
+  };
+
+  pythonImportsCheck = [ pname ];
```
```suggestion
  pythonImportsCheck = [ "euclid3" ];
```

SuperSandro2000 commented on pkgs/development/python-modules/euclid3/default.nix:
```diff
@@ -0,0 +1,24 @@
+
+  meta = with lib; {
+    description = "2D and 3D vector, matrix, quaternion and geometry module";
+    homepage = "https://pypi.org/project/euclid3/";
```
the real linked website is dead.

hesiod replied:
Do you mean I should add a comment about this? (the PyPI link works fine)

SuperSandro2000 replied:
Python packages which have no homepage or source are not that great

SuperSandro2000 commented:
@ofborg build python310Packages.pybars3 python310Packages.kikit python310Packages.pymeta3 python310Packages.kikit python310Packages.euclid3 python310Packages.pcbnew-transition

AndersonTorres commented on pkgs/development/python-modules/euclid3/default.nix:
```diff
@@ -0,0 +1,24 @@
+{ lib
+, buildPythonPackage
+, fetchPypi
+}:
```
```suggestion
}:

```

AndersonTorres commented on pkgs/development/python-modules/euclid3/default.nix:
```diff
@@ -0,0 +1,24 @@
+    maintainers = with maintainers; [ tmarkus ];
+  };
+}
+
```
```suggestion
```

AndersonTorres commented on pkgs/development/python-modules/pcbnew-transition/default.nix:
```diff
@@ -0,0 +1,33 @@
+, fetchPypi
+, kicad
+, versioneer
+}:
```
```suggestion
}:

```

AndersonTorres commented on pkgs/development/python-modules/pcbnew-transition/default.nix:
```diff
@@ -0,0 +1,33 @@
+    maintainers = with maintainers; [ tmarkus ];
+  };
+}
+
```
```suggestion
```

AndersonTorres commented on pkgs/development/python-modules/pybars3/default.nix:
```diff
@@ -0,0 +1,34 @@
+    maintainers = with maintainers; [ tmarkus ];
+  };
+}
+
```
```suggestion
```

AndersonTorres commented on pkgs/development/python-modules/pymeta3/default.nix:
```diff
@@ -0,0 +1,31 @@
+, buildPythonPackage
+, fetchPypi
+, twisted
+}:
```
```suggestion
}:

```

AndersonTorres commented on pkgs/development/python-modules/pymeta3/default.nix:
```diff
@@ -0,0 +1,31 @@
+    maintainers = with maintainers; [ tmarkus ];
+  };
+}
+
```
```suggestion
```

AndersonTorres commented on pkgs/development/python-modules/solidpython/default.nix:
```diff
@@ -0,0 +1,42 @@
+    maintainers = with maintainers; [ tmarkus ];
+  };
+}
+
```
```suggestion
```

hesiod commented:
Rebased on top of recent master along with the following changes:
* Update KiKit to 1.3.0 (previously 1.2.0)
* Update pcbnew-transition to 0.3.4 (previously 0.2.0)
* Add solidpython tests
* Add KiKit to `top-level/all-packages.nix` via `toPythonApplication` as it is a hybrid library/application
* Addressed whitespace/style issues from review comments

hesiod commented:
@ofborg build python310Packages.kikit python311Packages.kikit

hesiod commented:
@SuperSandro2000 If you have the time, perhaps you could clarify the two remaining review comments? Otherwise, I think this PR is pretty much good to go :)

r0hit05 commented:
We did not realise that a PR for kikit already exists. We ended up repeating a lot of this work over in #249464. Compared to this PR, it seems like we have added newer versions of certain dependencies. We have also added a kicad package set that can be used with a new `withPackages` helper on kicad.

Sorry for the redundant work!

jfly commented:
As https://github.com/NixOS/nixpkgs/pull/249464 has landed, I'm going to close this PR.

Again, sorry for the redundant work!
